superset/viz.py

```diff
@@ -1053,6 +1052,15 @@ def query_obj(self) -> QueryObjectDict:
         d = super().query_obj()
         fd = self.form_data
         d["metrics"] = fd.get("metrics")
+        mapping = {
+            "min": "PT1M",
+            "hour": "PT1H",
+            "day": "P1D",
+            "week": "P1W",
+            "month": "P1M",
+            "year": "P1Y",
+        }
+        d["extras"]["time_grain_sqla"] = mapping[fd.get("subdomain_granularity", "min")]
```

Comment:
Minor improvement proposal: To make this work for Legacy Druid, we'd need to set the `d["granularity"]` property to time grain. so something like this:
```python
time_grain = mapping[fd.get("subdomain_granularity", "min")]
if self.datasource.type == 'druid':
    d["granularity"] = time_grain
else:
    d["extras"]["time_grain_sqla"] = time_grain
```